system/datatypes/peer_reviewed_article.php

```diff
@@ -1117,6 +1243,10 @@ private function createCommentForm($type, array $values = [], HyphaDomElement $d
 			$vars['un-anonymous'] = __('art-comment-unanonymous');
 			$vars['email'] = __('art-email');
 			$vars['field-name-email'] = self::FIELD_NAME_DISCUSSION_COMMENTER_EMAIL . $dataFieldSuffix;
+			$vars['subscribe'] = __('art-subscribe');
+			$vars['info-button-subscribe'] = makeInfoButton('help-art-subscribe');
+			$vars['field-name-subscribe'] = self::FIELD_NAME_DISCUSSION_COMMENTER_SUBSCRIBE . $dataFieldSuffix;
+			$vars['field-value-subscribe-me'] = self::FIELD_VALUE_DISCUSSION_COMMENTER_SUBSCRIBE_ME;
```

Comment:
Why so much code over the checkbox value here? Just any non-empty value would be sufficient, and then in the handling just check to see if the value is set?

There is also some code to pass a value for the checkbox as part of `$formData`, which just sets the same value again, which also seems pointless to me? Why is all this needed?

Reply:
Good to know. And a good indication that the workings of the HTMLForm is obscure.
Removed redundant code.